Libraries/LibLine/Editor.cpp

```diff
@@ -226,11 +226,55 @@ void Editor::suggest(size_t invariant_offset, size_t static_offset, Span::Mode o
     m_suggestion_manager.set_suggestion_variants(internal_static_offset, internal_invariant_offset, 0);
 }
 
+void Editor::initialize()
+{
+    if (m_initialized)
+        return;
+
+    struct termios termios;
+    tcgetattr(0, &termios);
+    m_default_termios = termios; // grab a copy to restore
+    // Because we use our own line discipline which includes echoing,
+    // we disable ICANON and ECHO.
+    if (auto* term = getenv("TERM")) {
+        if (StringView { term }.starts_with("xterm"))
+            m_configuration.set(Configuration::Full);
+        else
+            m_configuration.set(Configuration::NoEscapeSequences);
+    } else {
+        m_configuration.set(Configuration::NoEscapeSequences);
+    }
```

Comment:
`StringView` can handle `nullptr` just fine, no need for the duplication here

Reply:
That's news to me.
much better :)